core/src/test/java/io/opentelemetry/android/OpenTelemetryRumBuilderTest.java

```diff
@@ -187,27 +192,43 @@ public void shouldBuildLogRecordProvider() {
         assertThat(payload.get(0)).isEqualTo(expected);
     }
 
-    @Ignore("To be updated once AndroidInstrumentation is fully implemented")
     @Test
     public void shouldInstallInstrumentation() {
-        OpenTelemetryRum.builder(application, buildConfig())
-                .addInstrumentation(
-                        instrumentedApplication -> {
-                            assertThat(instrumentedApplication.getApplication())
-                                    .isSameAs(application);
-                            //
-                            // instrumentedApplication.registerApplicationStateListener(listener);
-                            // TODO update with AndroidInstrumentation
-                        })
-                .build();
+        ServiceManager serviceManager = createServiceManager();
+        SessionIdTimeoutHandler timeoutHandler = mock();
+        AndroidInstrumentation localInstrumentation = mock();
+        AndroidInstrumentation classpathInstrumentation = mock();
+        AndroidInstrumentationRegistry.get().register(classpathInstrumentation);
 
-        verify(application).registerActivityLifecycleCallbacks(activityCallbacksCaptor.capture());
+        new OpenTelemetryRumBuilder(application, buildConfig(), timeoutHandler)
+                .addInstrumentation(localInstrumentation)
+                .build(serviceManager);
 
-        activityCallbacksCaptor.getValue().onActivityStarted(activity);
-        verify(listener).onApplicationForegrounded();
+        verify(serviceManager.getAppLifecycleService()).registerListener(timeoutHandler);
 
-        activityCallbacksCaptor.getValue().onActivityStopped(activity);
-        verify(listener).onApplicationBackgrounded();
+        verify(localInstrumentation).install(eq(application), notNull());
+        verify(classpathInstrumentation).install(eq(application), notNull());
+    }
+
+    @Test
+    public void shouldInstallInstrumentation_excludingClasspathImplsWhenRequestedInConfig() {
+        ServiceManager serviceManager = createServiceManager();
+        SessionIdTimeoutHandler timeoutHandler = mock();
+        AndroidInstrumentation localInstrumentation = mock();
+        AndroidInstrumentation classpathInstrumentation = mock();
+        AndroidInstrumentationRegistry.get().register(classpathInstrumentation);
+
+        new OpenTelemetryRumBuilder(
+                        application,
+                        buildConfig().disableInstrumentationDiscovery(),
+                        timeoutHandler)
+                .addInstrumentation(localInstrumentation)
+                .build(serviceManager);
+
+        verify(serviceManager.getAppLifecycleService()).registerListener(timeoutHandler);
+
+        verify(localInstrumentation).install(eq(application), notNull());
+        verifyNoInteractions(classpathInstrumentation);
```

Comment:
So I like this test because I think it is highlighting a current shortcoming with this approach....which I think is fine for now in this PR, but I'd like to see this addressed in the very near future. 

If I'm a user looking at this code for the first time, and I have a bit of manual instrumentation that I've created or initialized or whatever, it's unclear if I'm supposed to register it directly with the registry or if I should `addInstrumentation()` on the `OpenTelemetryRumBuilder`. A causal reader might think that these are two ways of solving the same problem...and they kind of are, until `discoverInstrumentations` gets into the mix. 

The configuration via `discovrerInstrumentations` just prevents the contents of the registry from being added to the final list of instrumentations. It doesn't actually prevent the discovery! I think that even though the `instrumentations` field on the `AndroidInstrumentationRegistryImpl` is lazy, any interactions with it will basically cause it to use the `ServiceLoader` to load all instrumentations on the classpath....but just won't ever install those. For the most part that should be fine, simply creating an instrumentation shouldn't have much overhead and definitely shouldn't have side effects without `install()`...but we have no guarantees.

One way I can think of handling this is to decide if the registry really needs to have the `register` method at all. Maybe we can remove that method and then rename `AndroidInstrumentationRegistry` (and impl) to `AutodetectedClasspathInstrumentations` or something similar.

Reply:
Opened #502 to track this and not block this work.

Reply:
This is a great point, we should definitely avoid this ambiguity for adding custom instrumentations.

> One way I can think of handling this is to decide if the registry really needs to have the register method at all. Maybe we can remove that method and then rename AndroidInstrumentationRegistry (and impl) to AutodetectedClasspathInstrumentations or something similar.

I think this sounds good.